var/spack/repos/builtin/packages/p7zip/package.py

```diff
@@ -14,6 +14,7 @@ class P7zip(MakefilePackage):
     version('16.02', sha256='5eb20ac0e2944f6cb9c2d51dd6c4518941c185347d4089ea89087ffdd6e2341f')
 
     patch('gcc10.patch', when='%gcc@10:', sha256='96914025b9f431fdd75ae69768162d57751413634622f9df1a4bc4960e7e8fe1')
+    patch('gcc11.patch', when='%gcc@11:', sha256='39dd15f2dfc86eeee8c3a13ffde65c2ca919433cfe97ea126fbdc016afc587d1')
```

Comment:
On a second thought, can you please add a brief comment above this directive to explain what the patch does and link to its provenance or a bug report if it exists?

Reply:
Added a comment and rebased on top of develop.

There is no more provenance than is present in the commit. I discovered the issue when p7zip failed to build under GCC 11, and wrote the patch to fix it.

There is also no bug report, since the upstream project is no longer maintained (last release was in 2016).